include/SGM.hpp

```diff
@@ -1,13 +1,103 @@
 #pragma once
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/CANDevice.hpp>
+#include <EVT/io/CANOpenMacros.hpp>
+#include <co_core.h>
+#include <dev/StrainGauge.hpp>
+
+#define NUM_GAUGES 4
+
+namespace IO = EVT::core::IO;
 
 namespace SGM {
 
 /**
- * This is an example of a class for a board
+ * Measures strain/stress on component using reading from ADC.
  */
-class SGM {
+class SGM : public CANDevice {
 public:
+    /**
+     * Constructor takes an array 4 strain gauges
+     *
+     * @param[in] gauges the strain gauges
+     */
+    explicit SGM(DEV::StrainGauge gauges[NUM_GAUGES]);
+
+    /**
+     * The node ID used to identify the device on the CAN network.
+     */
+    static constexpr uint8_t NODE_ID = 4;
+
+    /**
+     * Updates the voltage values in an array from the SGM object.
+     */
+    void process();
+
+    /**
+     * Get the device's node ID
+     *
+     * @return The node ID of the can device.
+     */
+    uint8_t getNodeID() override;
+
+    /**
+     * Get the number of elements in the object dictionary.
+     *
+     * @return The number of elements in the object dictionary
+     */
+    uint8_t getNumElements() override;
+
 private:
+    /**
+     * Stores the 4 strain gauges
```

Comment:
For one-line comments for variables, we often do this because it takes up less space. The way you had it is still correct, though
```suggestion
     /** Stores the 4 strain gauges */
```